api_end_end_testing/src/test/java/tds/user/api/tests/CreateUserTest.java

```diff
@@ -0,0 +1,247 @@
+        roleAssociations.add(new RoleAssociation("Administrator", "CLIENT", "98765"));
+        roleAssociations.add(new RoleAssociation("Administrator", "CLIENT", "98765"));
+
+        Header authHeader = new Header("Authorization", "Bearer " + accessToken);
+
+        UserInfo userInfo = new UserInfo(email, "Betsy", "Carson", "800-332-4747", roleAssociations);
+
+        ObjectMapper mapper = new ObjectMapper();
+        String jsonInString = null;
+
+        try {
+            jsonInString = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(userInfo);
+            System.out.println("jsonInString: " + jsonInString);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        given()
+            .contentType(ContentType.JSON)
+            .header(authHeader)
+            .body(userInfo)
+            .when()
+            .post("/rest/external/user")
+            .then()
+            .statusCode(204);
+    }
+
+    /*
+        Create invalid user
+        Testing HTTP POST of /rest/external/user, 400 bad request
+    */
+    @Test(dependsOnMethods = "updateUser")
+    public void invalidUser() {
+
+        RestAssured.baseURI = userUri;
+
+        List<RoleAssociation> roleAssociations = new ArrayList<RoleAssociation>();
+        roleAssociations.add(new RoleAssociation("Administrator", "CLIENT", "98765"));
+        roleAssociations.add(new RoleAssociation("Administrator", "CLIENT", "98765"));
+
+        Header authHeader = new Header("Authorization", "Bearer " + accessToken);
+
+        UserInfo userInfo = new UserInfo(email, "Betsy", "", null, roleAssociations);
+
+        ObjectMapper mapper = new ObjectMapper();
+        String jsonInString = null;
+
+        try {
+            jsonInString = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(userInfo);
+            System.out.println("jsonInString: " + jsonInString);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        given()
+            .contentType(ContentType.JSON)
+            .header(authHeader)
+            .body(userInfo)
+            .when()
+            .post("/rest/external/user")
+            .then()
+            .statusCode(400);
+    }
+
+    /*
+        Get user by email
+        Testing HTTP GET of /rest/external/user/{email}/details, 200 success return student info
+    */
+    @Test(dependsOnMethods = "invalidUser")
+    public void getUserByEmail() {
+        RestAssured.baseURI = userUri;
+
+        Header authHeader = new Header("Authorization", "Bearer " + accessToken);
+
+        System.out.println("endpoint: " + "/rest/external/user/" + email + "/details");
+
+        given()
+            .contentType(ContentType.JSON)
+            .accept(ContentType.JSON)
+            .header(authHeader)
+            .when()
+            .get("/rest/external/user/" + email + "/details")
+            .then()
+            .statusCode(200);
```

Comment:
Since this endpoint returns some data I think we should test to make sure it contains the JSON we expect.  We do this in other projects using [hamcrest](https://code.google.com/archive/p/hamcrest/wikis/Tutorial.wiki]).  For example, this endpoint returns:

```json
{
  "email" : "betsy.ross@example.com",
  "firstName" : "Betsy",
  "lastName" : "Carson",
  "phoneNumber" : "800-332-4747",
  "roleAssociations" : [ {
    "role" : "Administrator",
    "level" : "CLIENT",
    "entityId" : "98765"
  }
```

Taking that into account we can test the email by tweaking your RestAssured code to like this:

```java
        given()
            .contentType(ContentType.JSON)
            .accept(ContentType.JSON)
            .header(authHeader)
        .when()
            .get("/rest/external/user/" + email + "/details")
        .then()
            .statusCode(200)
            .body("email", equalTo("betsy.ross@example.com"));
```
            
That .body("email"...) line is inspecting the JSON response.  You can chain them along to check for multiple things in the JSON body payload.  Here is an [example](https://github.com/SmarterApp/TDS_SessionService/blob/develop/service/src/test/java/tds/session/web/endpoints/SessionControllerIntegrationTests.java#L32).